relay_test.go

```diff
@@ -532,12 +532,7 @@ func TestRelayInboundConnContext(t *testing.T) {
 			return context.WithValue(c, "foo", "bar")
 		})
 		svr := ts.NewServer(svrOpts)
-		testutils.RegisterFunc(svr, "echo", func(ctx context.Context, args *raw.Args) (*raw.Res, error) {
-			// Verify that the context passed into the handler inherits from the base context
-			// set by ConnContext
-			assert.Equal(t, "bar", ctx.Value("foo"), "Value unexpectedly different from base context")
-			return &raw.Res{Arg2: args.Arg2, Arg3: args.Arg3}, nil
-		})
+		testutils.RegisterEcho(svr, nil)
```

Comment:
we can also get rid of the usage of the `ConnContext` function when creating the server. We should actually take advantage of `ts.Server()`